### PR TITLE
Attempt to integrating scarf sdk with tailcall

### LINKED ISSUE #612: Integrate Scarf SDK on NPM
Integrate scarf sdk with `@tailcallhq/tailcall`.

**Additional Info**
https://www.npmjs.com/package/@scarf/scarf

### PR DESCRIPTION
**Pull request by Min2who**
/claim #612

fixed #612 

**Build & Testing:**

- [ ✅] I ran `cargo test` successfully.
- [✅ ] I have run `./lint.sh --mode=fix` to fix all linting issues raised by `./lint.sh --mode=check`.

**Checklist:**

- [ ] I have added relevant unit & integration tests.
- [ ] I have updated the documentation accordingly (if applicable).
- [✅ ] I have performed a self-review of my own code.

### CODE REVIEW

**Comment by Min2who** [comment]
@amitksingh1490 could you please review it

File: npm/gen-root.ts [hunk 1]
```diff
@@ -1,81 +1,82 @@
-import * as fs from "fs/promises"
```
Comment (meskill) [inline npm/gen-root.ts:1]: do you have any idea why is github showing that the whole file was changed although there are not much changes? have you changed something in formatting?

File: npm/post-install.js [hunk 2]
```diff
@@ -1,30 +1,33 @@
...
+}
+
+const pkg = `@tailcallhq/core-${platform}-${arch}${libc}`
+
+try {
+  // @ts-ignore
+  const { stdout, stderr } = await execa(`npm install ${pkg}@${version} --no-save`)
+  stderr ? console.log(stderr) : console.log(`Successfully installed optional dependency: ${pkg}`, stdout)
+  // Install Scarf SDK as part of the post-install process
+  const { scarfStdout, scarfStderr } = await execa('npm i --save @scarf/scarf')
```
Comment (meskill) [inline npm/post-install.js:29]: why do we need to install the package additionally on the postinstall step? Shouldn't it be already installed by npm as it is specified in the package.json of tailcall package?

Comment (Min2who) [reply npm/post-install.js:29]: It will ensure that the package is installed or should we remove it in the package.json file

Comment (meskill) [reply npm/post-install.js:29]: package manager should ensure that package is installed we don't need to to this manually 🙂

Comment (Min2who) [reply npm/post-install.js:29]: ohk i will fix it

Comment (meskill) [reply npm/post-install.js:29]: so, I meant that installing package from postinstall script is a manual action that doesn't rely on definition of dependencies in package.json. And if we don't have a specific reason to do so it'd be better to just specify it in package.json and leave the package manager to their work

Comment (Min2who) [reply npm/post-install.js:29]: ohk
so can i make a function in gen-root.ts that does this

Comment (meskill) [reply npm/post-install.js:29]: adding it to the `dependencies` list in gen-root should be enough

Comment (Min2who) [reply npm/post-install.js:29]: can you check my coomit

Comment (meskill) [reply npm/post-install.js:29]: and go back to the initial question: do we need to change anything in here in post-install.js?

**Comment by tusharmath** [comment]
@Min2who can you resolve the conflicts please.

**Comment by Min2who** [comment]
ohk doing it !

**Comment by Min2who** [comment]
I have removed merge conflicts

**Comment by Min2who** [comment]
> @Min2who can you resolve the conflicts please.

@tusharmath i have removed merge conflicts

**Comment by Min2who** [comment]
> @Min2who can you resolve the conflicts please.

@tusharmath I have resolve the merge conflicts

**Pull request closed without merging**